### DISCUSSION TITLE
Unique code based Profile declaration

### DISCUSSION THREAD

**Discussion by Prateek-slokam** [Sunbird-Ed/Community#43] [Feature request]
The following is an ideation for enhancement to the user profile declaration on boarding the user. Requesting to add and comment on the requirement. This also might be relating to other BBs just starting here as need to identify that too.

Problem/Need statement:
The need has come up from one of the Adopter, DIKSHA. The ask is that users should be able to just add the UDISE code (a unique code for the schools) to declare their location/entities upon logging in. This would ease the selection and process for the user, if the user is aware of the code. Currently, as a process, the users have to search through the dropdown list of each parameter (for eg. District, block, school) one by one to select the bottom most of them.
In the generic form this can just be a unique code for any set of parameters of the declarations. For eg. in the health sector it might translate to the unique code of hospital or healthcare center. The user would have to directly enter the code of that hospital and the other structure, like state, district would also auto-populate accordingly.
This would also cater to one of a minor problems which is the name of many entities are same and thus, unique identification of the entity is only possible through the such a code which would let user select the particular entity that they want to.

Solution:
The unique code field can be an addition to the current profile declaration structure to provide the user choice where they can choose to enter the code or proceed with step by step selection to bottom most layer. The code upon being submitted by the user should fill up the complete profile on the basis of that code.

@luckynvdu @kameshbhr @aishwaryashikshalokam @varsha-sl @rhwarrier @sharathkashyap

**Comment by rhwarrier** [comment]
This was a suggestion earlier as well - but the ground reality (when we dug deeper at that point) was that the expectation of a user knowing the UDISE ID/ filling it up without error turned out to be unrealistic. Also, may states are adopting different codes (we've had an instance of a state changing their school codes in Diksha recently) - hence, this may not be possible to be done in a standardised manner. Will let @kameshbhr come in with inputs.

**Reply by kameshbhr** [reply]
@rhwarrier You are right that many users like students, parents and even teachers may not know the school code. However, other personas such as Head teachers would possibly know it. The request is to provide entering of the code as an additional option - not replace the current flow.
A user who knows the code can enter the code and if the code is valid, the fields can be auto populated using the code. Someone who doesn't know the code can still go with the current flow of selecting state, district etc. and the school.

**Reply by pramodkvarma** [reply]
@kameshbhr if you are auto populating school details from any other system (like UDISE), you need to also worry about error handling. What happens if the school data in that system is incorrect? Will the user then be allowed to clear the code and manually enter? Editing fields with code being populated is a bad idea given it will defeat the purpose of pulling the data altogether. 

Btw, NDEAR is planning a new education institution registry. Why is effort spent now to integrate with analytics system like UDISE especially knowing very few users will use that option?
